### PR TITLE
Merge branches

### PR DESCRIPTION
**Pull request by jakirkham**
Combines all of the version branches back into a `master` branch and adds a matrix of CUDA versions to build this package for. Should simplify maintenance work that affects multiple versions (like including a missing library or similar).

<hr>

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes https://github.com/conda-forge/cudatoolkit-feedstock/issues/8

### CODE REVIEW

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render.

**Comment by jakirkham** [comment]
Trying to re-render locally, the following issue occurs.

```python
$ conda smithy rerender -c
INFO:conda_smithy.configure_feedstock:Downloading conda-forge-pinning-2020.11.08.20.44.44
INFO:conda_smithy.configure_feedstock:Extracting conda-forge-pinning to /var/folders/lw/whnppx9951323x027sfhl5mmglk2pt/T/tmputn2kz2d
Setting build platform. This is only useful when pretending to be on another platform, such as for rendering necessary dependencies on a non-native platform. I trust that you know what you're doing.
WARNING:conda_build.config:Setting build platform. This is only useful when pretending to be on another platform, such as for rendering necessary dependencies on a non-native platform. I trust that you know what you're doing.
Setting build arch. This is only useful when pretending to be on another arch, such as for rendering necessary dependencies on a non-native arch. I trust that you know what you're doing.
WARNING:conda_build.config:Setting build arch. This is only useful when pretending to be on another arch, such as for rendering necessary dependencies on a non-native arch. I trust that you know what you're doing.
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Adding in variants from /var/folders/lw/whnppx9951323x027sfhl5mmglk2pt/T/tmputn2kz2d/conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from /var/folders/lw/whnppx9951323x027sfhl5mmglk2pt/T/tmputn2kz2d/conda_build_config.yaml
Adding in variants from /Users/jkirkham/Developer/conda-forge/feedstocks/feedstocks/cudatoolkit/recipe/conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from /Users/jkirkham/Developer/conda-forge/feedstocks/feedstocks/cudatoolkit/recipe/conda_build_config.yaml
Traceback (most recent call last):
  File "/Users/jkirkham/miniconda/bin/conda-smithy", line 10, in <module>
    sys.exit(main())
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/conda_smithy/cli.py", line 613, in main
    args.subcommand_func(args)
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/conda_smithy/cli.py", line 419, in __call__
    self._call(args, tmpdir)
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/conda_smithy/cli.py", line 424, in _call
    configure_feedstock.main(
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/conda_smithy/configure_feedstock.py", line 1937, in main
    render_travis(env, config, forge_dir, return_metadata=True)
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/conda_smithy/configure_feedstock.py", line 1067, in render_travis
    return _render_ci_provider(
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/conda_smithy/configure_feedstock.py", line 610, in _render_ci_provider
    metas = conda_build.api.render(
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/conda_build/api.py", line 42, in render
    metadata_tuples = render_recipe(recipe_path, bypass_env_check=bypass_env_check,
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/conda_build/render.py", line 800, in render_recipe
    m = MetaData(recipe_dir, config=config)
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/conda_build/metadata.py", line 886, in __init__
    self.parse_again(permit_undefined_jinja=True, allow_no_other_outputs=True)
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/conda_build/metadata.py", line 961, in parse_again
    self.meta = parse(self._get_contents(permit_undefined_jinja,
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/conda_build/metadata.py", line 1554, in _get_contents
    rendered = template.render(environment=env)
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/Users/jkirkham/miniconda/lib/python3.8/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/Users/jkirkham/Developer/conda-forge/feedstocks/feedstocks/cudatoolkit/recipe/meta.yaml", line 120, in top-level template code
    "win": "451.82",
TypeError: 'NoneType' object is not callable
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Copyright (c) 2017, Continuum Analytics, Inc. All rights reserved.
 #
 # Distribution of the content approved by NVIDIA ( http://nvbugs/3052604 )
-{% set major_minor = "11.0" %}
+{% set major_minor = major_minor_ver %}
```
Comment (isuruf) [inline recipe/meta.yaml:6]: You might have to set a default here for early stages of rendering.

Comment (jakirkham) [reply recipe/meta.yaml:6]: Good point. Thanks Isuru! 😄

File: recipe/meta.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # Copyright (c) 2017, Continuum Analytics, Inc. All rights reserved.
 #
 # Distribution of the content approved by NVIDIA ( http://nvbugs/3052604 )
-{% set major_minor = "11.0" %}
+{% set major_minor = major_minor_ver|default("11.0") %}
```
Comment (jakirkham) [inline recipe/meta.yaml:6]: Added `default` to ensure something is here in the base case.

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/meta.yaml]: Added a `default` here ( https://github.com/conda-forge/cudatoolkit-feedstock/pull/31#pullrequestreview-526539482 ).

**Comment by jakirkham** [comment]
Re-rendering now proceeds without failing. However it either makes no changes or it remove the `ppc64le` build. I think this is happening as it ends up selecting some version for the re-rendering (maybe not even the default we provide) and depending on what version it picks it either removes `ppc64le` or keeps it. Would have also thought re-rendering like this would generate a matrix of version builds, which doesn't seem to be happening. Maybe they are being placed in one CI job? Or maybe there is still some issue remaining?

**Comment by jakirkham** [comment]
Looks like there are still issues. The Linux build just built 9.2 and no other versions.

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by isuruf** [comment]
https://github.com/jakirkham-feedstocks/cudatoolkit-feedstock/pull/1

**Comment by jakirkham** [comment]
Thanks Isuru! 😄 Have merged and re-rendered. Was wondering if it might be something like this.

**Review by kkraus14 — APPROVED** [review]

**Comment by jakirkham** [comment]
cc @scopatz (in case you have thoughts here 🙂)

**Review by mike-wendt — APPROVED** [review]

**Comment by jakirkham** [comment]
Any other thoughts on this? 🙂

**Review by leofang — APPROVED** [review]

**Comment by jakirkham** [comment]
Sound like we are ok with this change. Going to go ahead and merge. If there are any issues, please let us know. Happy to follow up in additional issues and PRs.

**Comment by jakirkham** [comment]
Have also gone back through and trimmed the version branches. All of their history is already included in the `master` branch as of this PR.

**Pull request merged**